include/ensmallen_bits/wn_grad/wn_grad.hpp

```diff
@@ -0,0 +1,133 @@
+/**
+ * @file wn_grad.hpp
+ * @author Marcus Edel
+ *
+ * WNGrad is a general nonlinear update rule for the learning rate.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_WN_GRAD_WN_GRAD_HPP
+#define ENSMALLEN_WN_GRAD_WN_GRAD_HPP
+
+#include <ensmallen_bits/sgd/sgd.hpp>
+#include "wn_grad_update.hpp"
+
+namespace ens {
+
+/**
+ * WNGrad is a general nonlinear update rule for the learning rate, so that
+ * the learning rate can be initialized at a high value, and is subsequently
+ * decreased according to gradient observations.
+ *
+ * For more information, see the following.
+ *
+ * @code
+ * @article{Wu2018,
+ *   author  = {{Wu}, X. and {Ward}, R. and {Bottou}, L.},
+ *   title   = {WNGrad: Learn the Learning Rate in Gradient Descent},
+ *   journal = {ArXiv e-prints},
```

Comment:
Looks to me like this is NIPS formatting---do you know if the paper got accepted?  Might be nicer to include the NIPS version.  (Although, I suppose the bibliographic reference is not ready yet since the conference has not happened.)

Reply:
Right, we should update the reference once there is something to update.

Reply:
NIPS seems to be rebranding to NeurIPS: https://neurips.cc/
Didn't realise the previous name was actually a problem until the rebranding.